src/netlink.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2019 Xaptum, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "netlink.h"
+
+#include <net/if.h>
+#include <fcntl.h>
+#include <unistd.h>
+
+int
+enftun_netlink_read_message(struct enftun_netlink* nl)
```

Comment:
Let's actually return the message here.  Make the signature
```
enftun_netlink_read_message(struct enftun_netlink* nl, char* buf, size_t buflen);
```

The caller (conn state) will have to provide a buffer, but can ignore the data returned in it.